### DISCUSSION TITLE
ScriptalbeObject 에서 list 로 많은 값을 가지고 싶어요!

### DISCUSSION THREAD

**Discussion by yoonjoohwan** [naverz/zepeto-studio-kor#630] [5. Q&A]
//test class
export default class test{
    public id:number;
    public name:string;
    public price:number;

    constructor(_id:number, _name:string , _price:number){
        this.id = _id;
        this.name = _name;
        this.price = _price;
    }
}
 
//testttttt class
import test from './test'
export default class testttttt{
    public List_test:Array<test> = new Array<test>();
}

![image](https://user-images.githubusercontent.com/91707090/163786678-31989aeb-2607-4fce-af4f-06cf37c5cd3f.png)
![image](https://user-images.githubusercontent.com/91707090/163786793-ad669f52-e5a5-43ce-a045-5d4c483efb44.png)

위의 Test class 의 구조로 testttttt 의 list를 만들어서 testclass 형식의 list를 활용하고 싶습니다

![image](https://user-images.githubusercontent.com/91707090/163786866-883dc634-3fb0-4f6b-a1c0-224c4ff75cb0.png)

제페토 가이드의 scriptableobject 에 써져있는 방법은 다사용해봤는데 list를 push 하는 과정에서 
cannot read property 'push' of undefined 라는 오류 메세지가 뜹니다
고수님들 알려주세요!!
.

**Comment by Deytooh** [comment]
안녕하세요 ~ Array 를 쓰기전에 declaration 해야돼요 :)

`myArray: MyClass[];

Awake(){
this.myArray = []; // 이거 한번만 해야돼요 ! :)
}

PushInMyArray(newItem MyClass){
this.myArray.push(newItem);
}
ClearMyArray(){
this.myArray.length = 0;
}
`

**Reply by yoonjoohwan** [reply]
혹시 저 코드가 들어가는 위치가 어딘지 알 수 있을까요?
